### PR TITLE
864-Windows-installer-issue-fix

### PR DESCRIPTION
**Pull request by henkdegroot**
Resolved the windows installer issue when space exists in the path

### CODE REVIEW

File: windows/installer.nsi [hunk 1]
```diff
@@ -118,8 +118,8 @@ Var bRunApp
     !define prefix "${DEPLOY_PATH}\${buildArch}"
     !tempfile files
 
-    ; Find target folders (Probably here's an issue with quoting. If ${prefix} contains spaces, the installer folders aren't created in the right way.)
-    !system 'cmd.exe /v /c "for /f "usebackq" %d in (`dir /b /s /ad "${prefix}"`) do \
+    ; Find target folders (#864 resolved by adding delims= to the for function. This prevents splitting path names when there is a spaces in the path)
```
Comment (ann0see) [inline windows/installer.nsi:121]: For isn't a function usually. It's a loop. Can you change the comment? Also there's a grammar mistake: "there is a spaces"

File: windows/installer.nsi [hunk 2]
```diff
@@ -415,8 +415,8 @@ FunctionEnd
     !system 'cmd.exe /v /c "for /r "${prefix}" %f in (*.*) do \
         @(set "_f=%f" && echo Delete "$INSTDIR\!_f:${prefix}\=!" >> "${files}")"'
 
-    ; Find target folders in reverse order to ensure they can be deleted when empty
-    !system 'cmd.exe /v /c "for /f "usebackq" %d in \
+    ; Find target folders in reverse order to ensure they can be deleted when empty (include delims= to prevents splitting path names when there is a spaces in the path)
```
Comment (ann0see) [inline windows/installer.nsi:418]: Similar grammar mistake.

**Review by ann0see — CHANGES_REQUESTED** [review]
Thank you. I'll test it soon. See my comments which are just small hints.

File: windows/installer.nsi (hunk 1 above)
Comment (henkdegroot) [reply windows/installer.nsi]: You are correct, for is a loop....however cmd help does indicate it is a command. So now fixed the typos and changed function to command.
Thank you for the review!

File: windows/installer.nsi (hunk 2 above)
Comment (henkdegroot) [reply windows/installer.nsi]: At least I was consistent :) or copy/paste the same....not sure if that is a good or a bad thing though. Updated now.

**Comment by ann0see** [comment]
Although it's not needed, but could you also please combine the installer and uninstaller fix into one commit and add Fixes #864 into the commit description?

File: windows/installer.nsi (hunk 2 above)
Comment (ann0see) [reply windows/installer.nsi]: Thanks!

**Review by ann0see — APPROVED** [review]
Built it in a path with one space (and also in one with multiple ones), and I can confirm it works now. Thank you very much for solving this issue. Hopefully there aren't any side effects from adding delims=?

Note to maintainers: https://github.com/jamulussoftware/jamulus/blob/3ef69509798876af912f796fa2eeeb5e2af3e01c/src/util.cpp#L441 since you're a new contributor, he needs to be added.

**Comment by henkdegroot** [comment]
@ann0see I checked the delims option and by not specifying a delimiter it ignores any. My first test was setting a specific delimiter but that would have introduced the same issue when that character is uses.

**Comment by henkdegroot** [comment]
> Although it's not needed, but could you also please combine the installer and uninstaller fix into one commit and add "Fixes #864" into the commit description?

I have combined into one commit now. Hoping I did it correct, first time I have done this.

**Comment by ann0see** [comment]
Yep. That's ok.

You can add a description to every commit (not only a message description), but that's not needed.

**Review by pljones — APPROVED** [review]
Approving untested -- the quoting does look _very_ peculiar and c-n-p to a .bat file got a complaint before it got as far as expanding the variables.  I'll take @ann0see's and @henkdegroot's word that it works in the NSI :)

**Comment by ann0see** [comment]
Just merged it. Thank you again! You can add yourself as contributor here: https://github.com/jamulussoftware/jamulus/blob/fdd0f02419fbaf20ea63890887d311a40b4ab2c9/src/util.cpp#L441

And could also open an issue on the documentation repo and state that the mentioning of the spaces issue can be removed from the compiling file.

**Comment by henkdegroot** [comment]
Just create my 2nd PR....to add myself to the contributors :) 

Will now create the issue on the documentation repo to get this updated.

**Comment by ann0see** [comment]
Thanks!

**Pull request merged**